### ISSUE TITLE
Better UX when device has Airplane mode enabled or no Internet access.

### ISSUE THREAD

**Issue by 13rac1** [roundware/roundware-android#40]
Enable Airplane mode with no data or Wifi. Start the App. User gets a greyed/disabled `Listen`, but not greyed/disabled `Speak` button. We need a popup message to say "Yo! I need the bits and bytes man" Also, `Speak` should be disabled too.

**Comment by hburgund** [comment]
I agree there should be some notification, but `Speak` is left enabled on purpose since there is "offline mode" that allows users to make recordings, store them locally and upload them later when they open the app with connectivity.
Of course, this feature may not be desired with FAM's project, but it's a very useful feature to have in certain situations.

**Comment by 13rac1** [comment]
@hburgund If that feature worked... Sure. Plus, the possibility needs to be clear to the user. Currently, pressing `Speak` results in a blank/white WebView in Airplane mode.

**Comment by hburgund** [comment]
Like I said, some notification is needed.  In iOS, I usually say something like "You have no connectivity, so you can't listen to streaming audio, but you can still make recordings if you want".  Try putting your iPhone 4 in Airplane mode and see what happens.

The feature did used to work on Android as well, but I guess it's no longer working for some reason.  Perhaps it's another one of those things that one of the long list of previous devs disabled for unknown reasons ;-(  Obviously, it isn't needed near-term, but it should be re-implemented at some point.

Agreed that the near-term fix is to disable both `Listen` and `Speak` and notify the user as to what's going on.

**Comment by 13rac1** [comment]
"You have no connectivity, so you can't listen to streaming audio, but you can still make recordings if you want" unless the webviews haven't been downloaded yet.

**Comment by multidynamic** [comment]
It may be cool to use one of these icons: http://google.github.io/material-design-icons/

**Comment by hburgund** [comment]
It is true that if the webview code hasn't been downloaded, it won't work.  But since it takes network connectivity to download the app in the first place, the vast majority of people are going to open the app after downloading, thereby getting the webview code and not encountering this problem in the future.  But I don't know if the Android app currently stores the webview files and only updates them when the version increments like it used to.  If it requires downloading the zip every time no matter what, then this is clearly a much more significant issue.

Either way, if someone downloads the app and doesn't open it right away, but waits to open it for the first time at a time when they have no network connectivity, there would be a problem.  For my iOS apps, I haven't worried about that edgy case, but there could of course be a series of different notifications or behaviors that are fully informative based on the specific scenario.  We could also package the default webview code for `Speak` in the apk, I suppose.

**Comment by 13rac1** [comment]
My preference (and therefore FAM's) is to notify the user of the situation and disable Speak functionality for the present. Long term, of course, I want people to be able to record without a network connection, but this feature needs so much work in it's current state.

**Comment by hburgund** [comment]
Agreed...this is not crucial core functionality for the first release and this alternative approach is totally acceptable.
